test/Transactions/Orleans.Transactions.Azure.Test/FaultInjection/FaultInjectionAzureTableTransactionStateStorage.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+using Orleans.Transactions.Abstractions;
+using Orleans.Transactions.AzureStorage;
+using Orleans.Transactions.Tests.FaultInjection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Orleans.Configuration;
+
+namespace Orleans.Transactions.Azure.Tests.FaultInjection
+{
+    public class FaultInjectionAzureTableTransactionStateStorage<TState> : ITransactionalStateStorage<TState>
+        where TState : class, new()
+    {
+        private readonly AzureTableTransactionalStateStorage<TState> stateStorage;
+        private readonly IControlledTransactionFaultInjector faultInjector;
+        public FaultInjectionAzureTableTransactionStateStorage(IControlledTransactionFaultInjector faultInjector,
+            AzureTableTransactionalStateStorage<TState> azureStateStorage)
+        {
+            this.faultInjector = faultInjector;
+            this.stateStorage = azureStateStorage;
+        }
+
+        public Task<TransactionalStorageLoadResponse<TState>> Load()
+        {
+            return this.stateStorage.Load();
+        }
+
+        public async Task<string> Store(
+
+            string expectedETag,
+            string metadata,
+
+            // a list of transactions to prepare.
+            List<PendingTransactionState<TState>> statesToPrepare,
+
+            // if non-null, commit all pending transaction up to and including this sequence number.
+            long? commitUpTo,
+
+            // if non-null, abort all pending transactions with sequence numbers strictly larger than this one.
+            long? abortAfter
```

Comment:
Usually doc comments like these would go above the method, like:

```xml
/// <summary>
/// Register a timer to send regular callbacks to this grain.
/// This timer will keep the current grain from being deactivated.
/// </summary>
/// <param name="asyncCallback"></param>
/// <param name="state"></param>
/// <param name="dueTime"></param>
/// <param name="period"></param>
/// <param name="name"></param>
/// <returns></returns>
```

You could include this info there.